### PR TITLE
Crosshair improvements and customization

### PR DESCRIPTION
**Pull request by VReaperV**
Adds a few cvars that allow customizing the crosshair colours.
cg_crosshairColor* sets the colour components for the crosshair.
cg_crosshairStyle sets whether to use the custom crosshair colour.

Adds optional crosshair outlines.
cg_crosshairOutlineColor* sets the colour components for the crosshair outline.
cg_crosshairOutlineStyle sets the crosshair outline colour mode:
0 - none (no outline)
1 - auto (colour is calculated as grayscale = crosshairColor.r * 0.299 + crosshairColor.g * 0.587 + crosshairColor.b * 0.114, grayscale += 0.5 * (grayScale < 0.5 ? : 1 : -1) if |0.5 - grayScale| < 0.25, otherwise 1 - grayScale).
2 - custom colour
cg_crosshairOutlineScale sets the outline scale (outline.xy = crosshair.xy - scale, outline.wh = crosshair.wh + scale * 2)

Crosshair images have been modified to have the correct alpha values and blend modes as well.

Old:
unvanquished_2024-02-08_021504_000
New (custom colour):
unvanquished_2024-02-08_021551_000
New (default colour, auto outline):
unvanquished_2024-02-08_021630_000
New (custom colour, auto outline):
unvanquished_2024-02-08_021646_000
New (custom colour, custom outline):
unvanquished_2024-02-08_021856_000

Old:
unvanquished_2024-02-08_022041_000
New:
unvanquished_2024-02-08_022029_000

### CODE REVIEW

File: src/cgame/cg_main.cpp [hunk 1]
```diff
@@ -53,6 +53,17 @@ Cvar::Range<Cvar::Cvar<int>> cg_drawClock("cg_drawClock", "draw clock (1 = 12-ho
 Cvar::Cvar<bool> cg_drawFPS("cg_drawFPS", "show client's frames per second", Cvar::NONE, true);
 Cvar::Range<Cvar::Cvar<int>> cg_drawCrosshair("cg_drawCrosshair", "draw crosshair (1 = ranged weapons, 2 = always)", Cvar::NONE, 2, 0, 2);
 Cvar::Cvar<bool> cg_drawCrosshairHit("cg_drawCrosshairHit", "show damage indicator", Cvar::NONE, true);
+Cvar::Range<Cvar::Cvar<int>> cg_crosshairStyle( "cg_crosshairStyle", "crosshair colour style (0 = image colour, 1 = custom colour)", Cvar::NONE, 0, 0, 1 );
```
Comment (DolceTriade) [inline src/cgame/cg_main.cpp:56]: Can this be an Enum for readability?

File: src/cgame/cg_rocket_draw.cpp [hunk 2]
```diff
@@ -763,8 +763,39 @@ class CrosshairHudElement : public HudElement {
...
+				crosshairColor.SetGreen( cg_crosshairColorGreen.Get() );
+				crosshairColor.SetBlue( cg_crosshairColorBlue.Get() );
+				crosshairColor.SetAlpha( cg_crosshairColorAlpha.Get() );
+			} else {
+				CG_GetRocketElementColor( crosshairColor );
+			}
+			if ( cg_crosshairOutlineStyle.Get() > 0 ) {
+				Color::Color outlineColor(crosshairColor);
+				if ( cg_crosshairOutlineStyle.Get() == 1 ) {
+					float grayScale = outlineColor.Red() * 0.299 + outlineColor.Green() * 0.587 + outlineColor.Blue() * 0.114;
```
Comment (DolceTriade) [inline src/cgame/cg_rocket_draw.cpp:778]: What are these magic numbers?

**Review by DolceTriade — COMMENTED** [review]
Looks good! just two basic questions!

Also (not blocking for this PR), would be cool if we could have a crosshair customizer in the menu.

File: src/cgame/cg_main.cpp (hunk 1 above)
Comment (DolceTriade) [reply src/cgame/cg_main.cpp:56]: I don't mean make the cvar an enum, but do comparisons against enums

File: src/cgame/cg_rocket_draw.cpp (hunk 2 above)
Comment (VReaperV) [reply src/cgame/cg_rocket_draw.cpp:778]: That is the grayscale formula using luma. I'll add a comment explaining that. I'm not sure if it's completely correct here with regards to linear/sRGB space, but seems to work alright.

Comment (DolceTriade) [reply src/cgame/cg_rocket_draw.cpp:778]: Ah. Yes, please leave a comment about this and also any deviations from the what is normally textbook so people know what's up.

File: src/cgame/cg_main.cpp (hunk 1 above)
Comment (VReaperV) [reply src/cgame/cg_main.cpp:56]: Done

File: src/cgame/cg_rocket_draw.cpp (hunk 2 above)
Comment (VReaperV) [reply src/cgame/cg_rocket_draw.cpp:784]: Done

**Review by DolceTriade — APPROVED** [review]

**Comment by VReaperV** [comment]
> Looks good!

Thanks!

> Also (not blocking for this PR), would be cool if we could have a crosshair customizer in the menu.

I'll see if I can add that later, though I've never used rmlUI before.

**Comment by DolceTriade** [comment]
> I'll see if I can add that later, though I've never used rmlUI before.

Definitely more a polish thing. Only do it if you find it fun 😄

**Comment by VReaperV** [comment]
Well, now we have it in the UI too :)

**Comment by illwieckz** [comment]
Very nice thing to add!

Instead of per-channel color cvars, why not a single color cvar with a `#RRGGBBAA` hex string?

**Comment by VReaperV** [comment]
That could work too, though I'm not sure if there's an easy way to change the UI for that. Also, console colours do the same thing.

**Comment by Grise3** [comment]
YEEEEHHAAAAA !
I will do the menus for that!
Thanks @VReaperV for this! Can change the crosshairs color is great!

**Comment by VReaperV** [comment]
No problem :) Old crosshair was really annoying me sometimes

**Comment by Grise3** [comment]
I did  a PR for change the crosshairs, for make them better and more visible.
 #2887
but I have to do some C++ code, and I don't code in C++ then I will have to take some hours to see what to change and how. @VReaperV I think that with all the crosshairs options ect, I will do a menu just for the crosshairs. :)

**Comment by DolceTriade** [comment]
@Grise3 If you ask for help with c++ code, we're happy to help you :)

**Comment by Grise3** [comment]
@DolceTriade oh, if youre okay for do it, it would be nice!
I don't understand the C++ of UNV XD
I have no idea if it is hard to do, or no

> If you ask for help with c++ code

I where not asking for help, but if you have time for this and if you're okay, It would be really nice!

**Comment by DolceTriade** [comment]
fwiw, I think that this is consistent with how other games do crosshair color. It also makes it easier to do menus since we can directly add range values to 3 cvars (otherwise you have to do fancy stuff with lua)

**Comment by Grise3** [comment]
> fwiw, I think that this is consistent with how other games do crosshair color. It also makes it easier to do menus since we can directly add range values to 3 cvars (otherwise you have to do fancy stuff with lua)

Yep make cars like this it's cool, but, I think that have the render (see the result I mean) would be nice too

**Comment by DolceTriade** [comment]
fwiw = For what it's worth

Also not to derail this PR, you can make a menu for this like this:

```html
<Crosshair/>
<input type="range" min="0" max="255" step="1" cvar="cg_crosshairColorRed" class="right"/>
<input type="range" min="0" max="255" step="1" cvar="cg_crosshairColorBlue" class="right"/>
<input type="range" min="0" max="255" step="1" cvar="cg_crosshairColorGreen" class="right"/>
```

**Comment by Grise3** [comment]
>fwiw = For what it's worth 

OK thx


>Also not to derail this PR, you can make a menu for this like this:

<Crosshair/>
<input type="range" min="0" max="255" step="1" cvar="cg_crosshairColorRed" class="right"/>
<input type="range" min="0" max="255" step="1" cvar="cg_crosshairColorBlue" class="right"/>
<input type="range" min="0" max="255" step="1" cvar="cg_crosshairColorGreen" class="right"/>

Haha yep

> @Grise3 If you ask for help with c++ code, we're happy to help you :) Just ping me directly, otherwise I don't always read all IRC backlog

 /ping @DolceTriade ^^

**Comment by VReaperV** [comment]
> @VReaperV I think that with all the crosshairs options ect, I will do a menu just for the crosshairs. :)

Nice. I've added a tab with the crosshair settings in this pr, but it was a quick change so any improvements are welcome (or redo that menu completely if you have a better overall idea for it).

**Comment by VReaperV** [comment]
> Yep make cars like this it's cool, but, I think that have the render (see the result I mean) would be nice too

I think so too, some sort of preview would be nice. For now though you can use the settings in-game so you can see the crosshair changing in the background.

**Comment by VReaperV** [comment]
I've fixed a small issue with aspect ratio not being taken into account, which came up during testing today. This is otherwise unchanged.

**Pull request merged**